lib/bolt/executor.rb

```diff
@@ -40,7 +40,7 @@ def initialize(concurrency = 1,
 
       @noop = noop
       @run_as = nil
-      @pool = Concurrent::ThreadPoolExecutor.new(max_threads: concurrency)
+      @pool = concurrency > 0 ? Concurrent::ThreadPoolExecutor.new(max_threads: concurrency) : nil
```

Comment:
I think if no concurrency is configured, we should use `Concurrent.global_immediate_executor`, which works like a normal Concurrent::Executor, but executes in the current thread.

Alternately, we could support passing *either* a Concurrent::Executor *or* a concurrency level here and let the caller decide. That would be useful if we wanted multiple Bolt::Executor instances backed by a single Concurrent::Executor

Reply:
I also wanted to avoid the promises, although maybe that's not much overhead.

Reply:
Alright, switched to using an ImmediateExecutor. I didn't see any negative impact in memory use from that (when running 1000 tasks).